doc/reference/reference_lua/box_null.rst

```diff
@@ -0,0 +1,83 @@
+.. _box-once:
+
+-------------------------------------------------------------------------------
+                             Constant `box.NULL`
+-------------------------------------------------------------------------------
+
+There are some major problems with using a Lua **nil** value in tuples.
+For example: you can't correctly assess length of sparse array (an array with
+nullable values set to null) in case of using Lua **nil** value.
+
+**Example:**
+
+.. code-block:: tarantoolsession
+
+    tarantool> t = {0, nil, 1, 2, nil}
+    ---
+    ...
+
+    tarantool> t
+    ---
+    - - 0
+      - null
+      - 1
+      - 2
+    ...
+
+    tarantool> #t
+    ---
+    - 4
+    ...
+
+To overcome this and some other issues use ``box.NULL`` constant instead.
+
+Using box.NULL
+--------------
+
+``box.NULL`` is a generic pointer to a C-like NULL. So it is
+some not **nil** value, even if it is a pointer to NULL.
+
+Use ``box.NULL`` only with capitalized NULL (``box.null`` is incorrect).
+
+.. NOTE::
+
+    Technically speaking ``box.NULL`` equals to ``ffi.cast('void *', nil)``.
+
+**Example:**
+
+.. code-block:: tarantoolsession
+
+    tarantool> t = {0, box.NULL, 1, 2, box.NULL}
+    ---
+    ...
+
+    tarantool> t
+    ---
+    - - 0
```

Comment:
typo

Reply:
This is a copy-paste from console. Not sure where the typo is

Reply:
![image](https://user-images.githubusercontent.com/2205188/80521740-35dc0c00-8994-11ea-957c-47344705c448.png)

Reply:
Indentation is broken

Reply:
i think i got it, thank you!